server/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java

```diff
@@ -363,6 +363,14 @@ static void init(
                             System.getProperty("java.home"));
             DeprecationLogger.getLogger(Bootstrap.class).deprecate("java_version_11_required", message);
         }
+        if (BootstrapInfo.getSystemProperties().get("es.xcontent.strict_duplicate_detection") != null) {
+            final String message = String.format(
+                Locale.ROOT,
+                "The Java option es.xcontent.strict_duplicate_detection is set to [%s]; " +
+                    "this option is deprecated and non-functional and should be removed from Java configuration.",
+                System.getProperty("es.xcontent.strict_duplicate_detection"));
```

Comment:
Why using System.getProperty here, but BootstrapInfo above?

Reply:
Good catch. Initially I used `System.getProperty` in both places, but the Forbidden APIs check only caught the use in the conditional. I'll make it consistent.